hangman.py

```diff
@@ -0,0 +1,200 @@
+        print("Good guess:", get_guessed_word(secret_word, letters_guessed))
+    else:
+        if letter in list_vowels:
+            guesses_point -= 2
+        else:
+            guesses_point -= 1
+        print("Oops! That letter is not in my word:", get_guessed_word(secret_word, letters_guessed))
+    return letters_guessed, guesses_point
+
+
+def hangman(secret_word, request, list_vowels, warnings_point, letters_guessed, guesses_point):
+    '''
+    Game Hangman. "request" show hangman(classic) or hangman with hints.
+    Print information for user, start loop, check user choose hangman or hangman with hints,
+    Check input letter and call letter_in_word, check if user guessed word.
+    '''
+    print("I am thinking of a word that is", len(secret_word), "letters long")
+    print("You have", warnings_point, "warnings left")
+    while guesses_point > 0:
+        print("-------")
+        print("You have", guesses_point, "guesses left")
+        print("Available letters:", end=" ")
+        get_available_letters(letters_guessed)
+        letter = input("Please guess a letter:")
+        if request is True and letter == "*":
+            show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+        else:
+            if letter not in string.ascii_letters or len(letter) != 1:
+                warnings_point, guesses_point = warnings("That is not a valid letter or you print more than one letter.", warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            elif letter in letters_guessed:
+                warnings_point, guesses_point = warnings("You have already guessed that letter:", warnings_point, guesses_point)
+                print(get_guessed_word(secret_word, letters_guessed))
+            else:
+                letters_guessed, guesses_point=letter_in_word(secret_word, letter, list_vowels, letters_guessed, guesses_point, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed) == 0:
+            final_points = guesses_point*len(set(secret_word))
+            print("------")
+            print("Congratulations, you won!")
+            print("Word:", secret_word)
+            print("Your total score for this game is:", final_points)
+            guesses_point = -1
+        elif is_word_guessed(secret_word, letters_guessed) != 0 and guesses_point <= 0:
+            print("------")
+            print("Sorry, you ran out of guessed. The word was", secret_word)
+            break
+
+
+def match_with_gaps(my_word, other_word):
+    '''
+    my_word: string with _ characters, current guess of secret word
+    other_word: string, regular English word
+    returns: boolean, True if all the actual letters of my_word match the
+        corresponding letters of other_word, or the letter is the special symbol
+        _ , and my_word and other_word are of the same length;
+        False otherwise:
+    '''
+    guesses_word = my_word.replace(" ", "")
```

Comment:
Could you have that replacement on 1 level up? to avoid performing replacement for same word many times